sql/moz-fx-data-shared-prod/stripe/itemized_payout_reconciliation/view.sql

```diff
@@ -45,34 +64,96 @@ charge_states AS (
   WHERE
     cards.country IN ("US", "CA")
 ),
+subscriptions AS (
+  SELECT
+    subscription_id,
+    plan_id,
+    plan_name,
+    plan_interval,
+    plan_interval_count,
+    product_id,
+    product_name,
+  FROM
+    `moz-fx-data-shared-prod`.subscription_platform.stripe_subscriptions
+),
 enriched AS (
   SELECT
-    report.*,
-    charge_states.state,
-    charge_states.postal_code,
+    report.* EXCEPT (
+      card_country,
+      shipping_address_city,
+      shipping_address_country,
+      shipping_address_line1,
+      shipping_address_line2,
+      shipping_address_postal_code,
+      shipping_address_state
+    ),
+    {% for country, state, postal_code, dst_pre in [
+        (
+          "customers.shipping_address_country",
+          "customers.shipping_address_state",
+          "customers.shipping_address_postal_code",
+          "shipping_address_"
+        ),
+        ("customers.country", "customers.state", "customers.postal_code", "address_"),
+        ("card_country", "charge_states.state", "charge_states.postal_code", "card_"),
+    ] -%}
+      CASE
+      -- american samoa
+        WHEN {{country}} ="US"
+          AND REGEXP_CONTAINS({{postal_code}}, "^96799(-?[0-9]{4})?$")
+          THEN STRUCT("AS" AS {{dst_pre}} country, NULL AS {{dst_pre}} state)
+      -- puerto rico
+        WHEN {{country}} ="US"
+          AND REGEXP_CONTAINS({{postal_code}}, "^00[679][0-9]{2}(-?[0-9]{4})?$")
+          THEN STRUCT("PR" AS {{dst_pre}} country, NULL AS {{dst_pre}} state)
+      -- virgin islands
+        WHEN {{country}} ="US"
+          AND REGEXP_CONTAINS({{postal_code}}, "^008[0-9]{2}(-?[0-9]{4})?$")
+          THEN STRUCT("VI" AS {{dst_pre}} country, NULL AS {{dst_pre}} state)
+        ELSE STRUCT({{country}} AS {{dst_pre}} country, {{state}} AS {{dst_pre}} state)
+      END.*,
+      {{postal_code}} AS {{dst_pre}} postal_code,
```

Comment:
Some odd formatting here, maybe caused by `bqetl format`?  In particular, the aliases containing spaces like `{{dst_pre}} country` won't be valid SQL.
```suggestion
      CASE
        -- American Samoa
        WHEN {{country}} = "US"
          AND REGEXP_CONTAINS({{postal_code}}, "^96799(-?[0-9]{4})?$")
          THEN STRUCT("AS" AS `{{dst_pre}}country`, NULL AS `{{dst_pre}}state`)
        -- Puerto Rico
        WHEN {{country}} = "US"
          AND REGEXP_CONTAINS({{postal_code}}, "^00[679][0-9]{2}(-?[0-9]{4})?$")
          THEN STRUCT("PR" AS `{{dst_pre}}country`, NULL AS `{{dst_pre}}state`)
        -- Virgin Islands
        WHEN {{country}} = "US"
          AND REGEXP_CONTAINS({{postal_code}}, "^008[0-9]{2}(-?[0-9]{4})?$")
          THEN STRUCT("VI" AS `{{dst_pre}}country`, NULL AS `{{dst_pre}}state`)
        ELSE STRUCT({{country}} AS `{{dst_pre}}country`, {{state}} AS `{{dst_pre}}state`)
      END.*,
      {{postal_code}} AS `{{dst_pre}}postal_code`,
```

Reply:
yep, caused by `bqetl format`